zerver/views/development/registration.py

```diff
@@ -92,8 +95,11 @@ def register_demo_development_realm(request: HttpRequest) -> HttpResponse:
     email = f"{name}@zulip.com"
     realm_name = generate_demo_realm_name()
     realm_type = Realm.ORG_TYPES["business"]["id"]
-    prereg = create_preregistration_user(email, None, realm_creation=True, password_required=False)
-    activation_url = create_confirmation_link(prereg, Confirmation.REALM_CREATION)
+    realm_subdomain = realm_name
+    prereg = create_preregistration_realm(email, realm_name, realm_subdomain, realm_type)
```

Comment:
These should probably be named `prereg_realm` rather than carrying over the old mistake of naming `prereg` instead of the usual `prereg_user` convention?

Reply:
Updated it. We still need to rename `prereg` to `prereg_user` for other existing cases. Will do it as a follow-up.